### PR TITLE
Extract execution creation into ExecutionFactory

### PR DESCRIPTION
**Pull request by bratseth**
@arnej27959 please review

### CODE REVIEW

**Comment by bratseth** [comment]
Merging to get a test run with this.

File: application/src/main/java/com/yahoo/application/Application.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
     /**
      * This system property is set to "true" upon creation of an Application.
-     * This is useful for components which are created by dependendy injection which needs to modify
+     * This is useful for components which are created by dependendcy injection which needs to modify
```
Comment (arnej27959) [inline application/src/main/java/com/yahoo/application/Application.java]: dependendcy -> dependency

File: application/src/main/java/com/yahoo/application/container/JDisc.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
 import com.yahoo.jdisc.test.TestDriver;
 import com.yahoo.processing.handler.ProcessingHandler;
 import com.yahoo.search.handler.SearchHandler;
+import com.yahoo.search.searchchain.ExecutionFactory;
```
Comment (arnej27959) [inline application/src/main/java/com/yahoo/application/container/JDisc.java]: ExecutionFactory not used here?

**Review by arnej27959 — COMMENTED** [review]
the ExecutionFactory is only used in SearchHandler now, should it be used more places?
two very minor nits.
overall looks good!

**Comment by bratseth** [comment]
Thanks, nits fixed in https://github.com/vespa-engine/vespa/pull/9745

> the ExecutionFactory is only used in SearchHandler now, should it be used more places?

Not afaik, but some applications wants to do this.

**Pull request merged**